clients/drcachesim/tracer/tracer.cpp

```diff
@@ -1687,8 +1687,8 @@ fork_init(void *drcontext)
 DR_EXPORT void
 drmemtrace_client_main(client_id_t id, int argc, const char *argv[])
 {
-    /* We need 2 reg slots beyond drreg's eflags slots => 3 slots */
-    drreg_options_t ops = { sizeof(ops), 3, false };
+    /* drreg will add 1 slot for eflags internally */
```

Comment:
I have some concerns as getting this number wrong can cause subtle performance issues and we have no regression tests for that:

- Are we sure this smaller number is safe: what does drreg_max_slots_used say (on a non-trivial app for coverage)
- The docs for the number of slots still say to add 1
- What about all the other drreg-using clients: are they adding 1 (I thought they were)
- Why didn't the commit that added 1 internally change the uses?  Did it change some uses?

Reply:
I do not think anything needs to be changed here.  The "1 slot for eflags internally" is really for the xax register:
```
        /* We get an extra slot for aflags xax, rather than just documenting that
         * clients should add 2 instead of just 1, as there are many existing clients.
         */
```
Clients still need to add 1, as the docs say.